dev/plugins/hu.elte.txtuml.api.model.execution/src/hu/elte/txtuml/api/model/execution/impl/seqdiag/SeqDiagModelExecutorThread.java

```diff
@@ -80,7 +82,21 @@ private void compareToPattern(SignalWrapper wrapper, ModelClass target, boolean
 		 * automatically accepted. Otherwise, we show an error.
 		 */
 		if (!result && mode == ExecMode.STRICT) {
-			root.getExecutor().addError(new InvalidMessageError(actual));
+			root.getExecutor().addError(new InvalidMessageError(actual, ErrorLevel.ERROR));
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Verifies that the {@code ModelClass} instance is currently in the given
+	 * state. If the assertion fails, a {@link ModelStateAssertError} will be
+	 * added.
+	 */
+	public void assertState(ModelClass instance, Class<?> state) {
+		Class<?> currentState = ((SeqDiagModelClassRuntime) ((DefaultSeqDiagRuntime) getModelRuntime())
```

Comment:
I suggest to break up these lines to named subexpressions by introducing temporary variables. Also, are these casts really necessary? That is, couldn't they be abstracted away with appropriate interface changes?